os/fs/driver/mtd/smart.c

```diff
@@ -5453,48 +5451,49 @@ static int smart_journal_recovery(FAR struct smart_struct_s *dev, journal_log_t
 
 	fvdbg("address : %u\n", address);
 	type = GET_JOURNAL_TYPE(log->status);
-	/* TODO Recovery Logic Will be added, after recovery we should update dev structure too(cf.freecount) */
+	
 	switch (type) {
-	case SMART_JOURNAL_TYPE_COMMIT:
 		
-		break;
+	case SMART_JOURNAL_TYPE_COMMIT:
+		/* Journal is valid, so we will verify written data on data sector with crc of MTD header in journal */
+		memcpy(dev->rwbuffer, &log->mtd_header, mtd_size);
+		ret = MTD_READ(dev->mtd, psector * dev->sectorsize + mtd_size, dev->sectorsize - mtd_size, (FAR uint8_t *)&dev->rwbuffer[mtd_size]);
+		if (ret != dev->sectorsize - mtd_size) {
+			fdbg("Read from data sector error : %d\n", ret);
+			return -EIO;
+		}
 		
-	case SMART_JOURNAL_TYPE_RELEASE:
+		ret = smart_validate_crc(dev);
+		/* If CRC is not same, data sector is invalid, so release data sector First */
+		if (ret != OK) {
+			fdbg("Invalid data, release dada sector & journal!\n");
```

Comment:
Please change `dada` to `data`.

Reply:
Thanks but due to refactor of adjust count values in smart_struct(c.f dev->erasecount[block]), I will change it next PR.